File: fibonacci.rb
def fibonacci(num)
  return num if (num < 2) 

  fibo_one, fibo_two = [0, 1]

  (num - 1).times do
    fibo_one, fibo_two = fibo_two, fibo_one + fibo_two
  end

  fibo_two
end


# pseudo-code
# input n
# if n=1, output=1
# if n=2, output=1
# if n>2 , sum of previous two values


if __FILE__ == $PROGRAM_NAME
  puts "Expecting: 0"
  puts "=>", fibonacci(0)

  puts

  puts "Expecting: 1"
  puts "=>", fibonacci(2)

  puts

  puts "Expecting: 55"
  puts "=>", fibonacci(10)

  # Don't forget to add your own!
end

# Please add your pseudocode to this file
# And a written explanation of your solution

File: fibonacci.js
function fibonacci(num) {
  if (num < 2) return num;

  let fibo = [0, 1];

  for (let i = 0; i < num - 1; i++){
    let sum = fibo[0] + fibo[1];
    fibo = [fibo[1], sum]
  }
  return fibo[1]
}



if (require.main === module) {
  // add your own tests in here
  console.log("Expecting: 0");
  console.log("=>", fibonacci(0));

  console.log("");

  console.log("Expecting: 1");
  console.log("=>", fibonacci(2));

  console.log("");

  console.log("Expecting: 55");
  console.log("=>", fibonacci(10));

  console.log("Expecting: 21");
  console.log("=>", fibonacci(8));

  console.log("Expecting: 233");
  console.log("=>", fibonacci(13));
}

module.exports = fibonacci;

// Please add your pseudocode to this file
// And a written explanation of your solution
